src/data-card-advanced.tsx

```diff
@@ -0,0 +1,498 @@
+                                {pretitle && (
+                                    <Text2
+                                        aria-label="Texto"
+                                        color={vars.colors.textPrimary}
+                                        truncate={pretitleLinesMax}
+                                        as={pretitleAs}
+                                        regular
+                                        hyphens="auto"
+                                    >
+                                        {pretitle}
+                                    </Text2>
+                                )}
+                                <Text
+                                    aria-label="Texto"
+                                    color={vars.colors.textPrimary}
+                                    mobileSize={20}
+                                    mobileLineHeight="28px"
+                                    desktopSize={20}
+                                    desktopLineHeight="28px"
+                                    truncate={titleLinesMax}
+                                    weight={textPresets.cardTitle.weight}
+                                    as={titleAs}
+                                    hyphens="auto"
+                                >
+                                    {title}
+                                </Text>
+                                <Text2
+                                    aria-label="Texto"
+                                    color={vars.colors.textPrimary}
+                                    truncate={subtitleLinesMax}
+                                    as="p"
+                                    regular
+                                    hyphens="auto"
+                                >
+                                    {subtitle}
+                                </Text2>
+                            </Stack>
+                        </Stack>
+                    </header>
+                )}
+
+                {description && (
+                    <Text2
+                        truncate={descriptionLinesMax}
+                        as="p"
+                        regular
+                        color={vars.colors.textSecondary}
+                        hyphens="auto"
+                        aria-label="Texto"
+                    >
+                        {description}
+                    </Text2>
+                )}
+            </Stack>
+        </div>
+    );
+};
+
+type CardFooterProps = {
+    button?: RendersNullableElement<typeof ButtonPrimary>;
+    footerImage?: RendersNullableElement<typeof Image>;
+    footerText?: string;
+    footerTextLinesMax?: number;
+    buttonLink?: RendersNullableElement<typeof ButtonLink>;
+};
+
+const CardFooter: React.FC<CardFooterProps> = ({
+    button,
+    footerImage,
+    footerText,
+    footerTextLinesMax,
+    buttonLink,
+}) => {
+    const {isMobile, isTabletOrBigger} = useScreenSize();
```

Comment:
try to use mediaqueries in css.ts instead of this hook. Mediaqueries are better for SSR support

Reply:
The isTabletOrBigger can be changed by putting it in the css, but not the isMobile, because it is used as a validator

Reply:
you can do the same with css classes and mediaqueries. Add a class when `hasAllItems` is true, then add some mediaqueries to that class.